docs/extensions/shoot-maintenance.md

```diff
@@ -0,0 +1,10 @@
+# Shoot maintenance
+Shoots configure a maintenance time window in which Gardener performs maintenance related tasks like infrastructure reconciliation, updating K8s or machine image versions.
+
+## Restart control plane controllers
+It is possible to configure Gardener to restart/delete certain control plane pods during a shoot's maintenance.
+Stale states and caches are refreshed and thus solve potential dead-locks or starving routines.
+
+Gardener, for example, takes this precautionary measure for `Kube-Controller-Manager` pods. 
+
+If your Gardener extension manages deployments of a shoot's control plane (shoot namespace in seed) and it could potentially profit from a regular restart please consider labeling your pods with `maintenance.gardener.cloud/restart: true`.
```

Comment:
Generally, this applies for pods, not deployments. It would also work for Statefulset/other workload resources.